### PR TITLE
[bitnami/jupyterhub] feat!: :lock: :boom: Improve security defaults

### PR DESCRIPTION
**Pull request by javsalgar**
BREAKING CHANGE

Signed-off-by: Javier Salmeron Garcia <jsalmeron@vmware.com>

### Description of the change

This major bump changes the following security defaults:

- `runAsGroup` is changed from `0` to `1001`
- `readOnlyRootFilesystem` is set to `true`
- `resourcesPreset` is changed from `none` to the minimum size working in our test suites (NOTE: `resourcesPreset` is not meant for production usage, but `resources` adapted to your use case).
- `global.compatibility.openshift.adaptSecurityContext` is changed from `disabled` to `auto`.

### Benefits

- Better compliance with NSA and MITRE security checklists
- Improved compatibility with Openshift

### Possible drawbacks

This could potentially break any customization or init scripts used in your deployment. If this is the case, change the default values to the previous ones.

### Applicable issues

- https://github.com/bitnami/charts/issues/24251

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by fmulero — COMMENTED** [review]
Thanks a lot @javsalgar 

I left some questions but I had a general one regarding to the readonly filesystem. Now the containers supporting the notebooks allow writes on `/opt/bitnami/jupyterhub-singleuser` Shouldn't be set the HOME variable in the container to that path?

File: bitnami/jupyterhub/templates/image-puller/networkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  {{- $podLabels := include "common.tplvalues.merge" ( dict "values" ( list .Values.imagePuller.podLabels .Values.commonLabels ) "context" . ) }}
+  podSelector:
+    matchLabels: {{- include "common.labels.matchLabels" ( dict "customLabels" $podLabels "context" $ ) | nindent 6 }}
+      app.kubernetes.io/component: image-puller
+  policyTypes:
+    - Ingress
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/jupyterhub/templates/image-puller/networkpolicy.yaml:23]: If we don't add an Ingress rule then all ports would be accessible. We add an empty Ingress to not allow any ingress access to the pod.

File: bitnami/jupyterhub/templates/image-puller/daemonset.yaml [hunk 2]
```diff
@@ -129,13 +129,31 @@ spec:
...
           {{- end }}
           {{- if .Values.imagePuller.customStartupProbe }}
-          livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.imagePuller.customStartupProbe "context" $) | nindent 12 }}
+          startupProbe: {{- include "common.tplvalues.render" (dict "value" .Values.imagePuller.customStartupProbe "context" $) | nindent 12 }}
+          {{- else if .Values.imagePuller.startupProbe.enabled }}
+          startupProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.proxy.startupProbe "enabled") "context" $) | nindent 12 }}
+            exec:
+              command:
+                - pgrep
+                - sleep
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/jupyterhub/templates/image-puller/daemonset.yaml:138]: Yes, as this pod is meant for pulling the images beforehand in all nodes

**Comment by javsalgar** [comment]
Agree on changing the HOME variable. We would need to update the jupyterhub container.

**Comment by fmulero** [comment]
> Agree on changing the HOME variable. I checked it is already set in the container
> 
> https://github.com/bitnami/containers/blob/main/bitnami/jupyter-base-notebook/4/debian-12/Dockerfile#L18

Looks good to me, but there is something weird. If we log into the jupyterhub console, the provided environment is opened by default in `/`, is not using `HOME` variable. It seems a question for jupyterhub project. Thanks @javsalgar for checking it.

**Review by fmulero — APPROVED** [review]
LGTM

**Review by fmulero — DISMISSED** [review]

**Review by alemorcuq — APPROVED** [review]

**Pull request merged**